### ISSUE TITLE
Calendar Settings: Admin to define calendar settings and search and filter options

### ISSUE THREAD

**Issue by troughc** [culturecreates/footlight-app#635]
### New Feature
Admin can configure calendar for their use in Settings. There are 2 new tabs for this: Calendar settings and Mandatory fields. In "Calendar settings" (renamed from Calendar metadata, they can:
- do basic calendar setup.
- set up the calendar widget to link back to their site.
-  add filters to listing pages, to create more refined views in different workspaces. 

In Mandatory fields, they can:
- set mandatory/required fields in forms.

To add filters to listing pages, the admin needs to be able to make taxonomies available as filters. These Admin-selected filters appear at the top of listing pages (as they do today). 
Examples of what an Admin may want to do: 
- Add a filter to the Places workspace that allows users to display only their places (or places created by someone else)
- Add a filter for types of places, regions, etc. 

Figmas for these changes:

1. Complete screens for calendar settings (excluding mobile and tablet views):
https://www.figma.com/file/DBnSSMT9Etfm6guDKKSp8O/Culture-Creates---Footlight-Calendar?type=design&node-id=4205%3A98575&mode=design&t=pApeEFryjppmXjMt-1
2. In taxonomies, a new toggle has been added so that the Admin can make specific taxonomies available as filters: https://www.figma.com/file/DBnSSMT9Etfm6guDKKSp8O/Culture-Creates---Footlight-Calendar?type=design&node-id=5145%3A85568&mode=design&t=t6hNFoQ93byOVOGv-1

Mobile and tablet views are covered by [this issue](https://github.com/culturecreates/footlight-app/issues/940)

cc @AbhishekPAnil
I am pausing this issue and starting to work in https://github.com/culturecreates/footlight-calendar-api/issues/780 .I need migrate from formschema to formfields in event form first,otherwise I would have to handle both forms and formschema in the mandatory fields section.

**Comment by troughc** [comment]
@sahalali @saumier should any fields be hard coded mandatory? For example, in Events edit form, should Title, Date, and Location be hardcoded mandatory (not configurable by the Admin in the calendar settings)?

**Comment by sahalali** [comment]
@troughc I didn't follow you. What do you mean by hardcoding fields?
The event form is not based on the data-centric approach. All fields are currently hardcoded in the UI.

**Comment by troughc** [comment]
@sahalali I mean, in the Footlight CMS, some fields should always be mandatory in every calendar, for every customer (so they wouldn't be user defined/configurable). In the calendar settings section of the Footlight CMS, these properties will be displayed in the UI as mandatory fields and the user cannot modify or delete them.
  
For example, on the event form, Title, date, and location could ALWAYS be mandatory as these properties are considered to be minimal requirements for an event to be whole.

My opinion is that, to achieve this, every calendar would come preconfigured with certain mandatory fields (for each entity). This is my suggested list. @tammy-culture @saumier can you please review and sign off if you agree (or share feedback).
### Required fields for all footlight CMS calendars (pre-configured before customer configuration/customization)
**Event**: Event title, date, location
**Place**: Place name, address
- **Address** minimal requirements: street address, postal code



**Organization**: Name
**Person**: Name
**Taxonomy and Settings**: As is. User doesn't configure mandatory fields for these 

thanks.

**Comment by troughc** [comment]
@SuhailAliyar please add a link to a general template document here where calendars are preconfigured with recommended settings (both mandatory and default/recommended that can be modified by a user). For example, settings for (what I can think of) :
- image aspect ratio
- taxonomy vocabularies
- extra recommended mandatory settings
- filters that are always there in different calendars (or is this necessary?).....

**Comment by sahalali** [comment]
@troughc I take a look into calendar creation today.

We can create a calendar with a name, 'namespace', and image. (namespace is no longer required.)
Currently, we do not have a configuration to use as a template for creating a calendar.  We need to create one.
The template would include:
1. image config : The max width and aspectRatio configuration of different types of image under a form. 
2. forms: The form configuration supports dynamic data-centric form creation.
3. language-fallbacks: Language fallback configuration
4. formSchema: For the event form. (This will be moved to forms later)
5. taxonomy vocabularies:

**Comment by troughc** [comment]
@sahalali here is a link to the figma for calendar settings. Also, the minimum requirements that an Admin cannot change or edit are listed in the previous comment.
: https://www.figma.com/file/DBnSSMT9Etfm6guDKKSp8O/Culture-Creates---Footlight-Calendar?type=design&node-id=4205-98575&mode=dev

Is there any other action you need from me?

**Comment by troughc** [comment]
@sahalali @saumier I have a question regarding the required calendar widget settings that will be displayed in this section of the UI (the links (URL templates). These links link the search results from the calendar widget back to the listing site that is providing the widget to their partners). For the time being, do we need these settings to be displayed in the UI for us to push the calendar widget feature to production? 
I _think_ we have all the required information collected in the backend for our two customers. Can you please confirm?
In other words, can we push the widget settings first and then add the calendar settings later?

**Comment by sahalali** [comment]
Yes, we can keep the config using backend in the database.

**Comment by troughc** [comment]
> Yes, we can keep the config using backend in the database.

@sahalali let's do this in two steps then: 
- first you add these configs to the back end in the database (for Signe Laval and for Tout Culture) to be ready when we push the Calendar widget to production. I will open this as a separate issue to keep it clear.
- Second we schedule the rest of the work through this issue.

**Comment by sahalali** [comment]
@sahalali to prepare a proposal and estimate it.

**Comment by AbhishekPAnil** [comment]
Frontend estimates: 42h
1.Calendar settings page with api integration : 22h
2.Toggle for filter in taxonomy : 4h
3.Mandatory fields  (Assuming separate api integration and depends on the api) : 16h

**Comment by sahalali** [comment]
Backend estimation: Total: 30H

1. Add support to add and update the following fields under Calendar - 2H
     a. updating image field to logo field
     b. adding dateFormatDisplay
     c. adding timezone
     d. adding widgetSettings (listEventsUrlTemplate,eventDetailsUrlTemplate)
     e. adding filterPersonalization(Event,Place,Organization,Person)
2. Add support to add./update taxonomy - 2H
    a. add field addToFilter to calendarDTO
3. Create a calendar settings template, to have a few default settings when a super admin creates a new calendar - 6H
4. Create a calendar with the calendar template to have minimum configurations. - 4H
5. Update event listing API to filter by taxonomies selected to "addToFilter" option and as per the filter personalization in calendar settings. - 4H
6. Update place listing API to filter by taxonomies selected to "addToFilter" option and as per the filter personalization in calendar settings. - 4H
7. Update organization listing API to filter by taxonomies selected to "addToFilter" option and as per the filter personalization in calendar settings. - 4H
8. 7. Update person listing API to filter by taxonomies selected to "addToFilter" option and as per the filter personalization in calendar settings. - 4H

**Comment by troughc** [comment]
@SuhailAliyar please see issue #798 for an update on the read only mode implementation. There is an option in Settings.

**Comment by AbhishekPAnil** [comment]
cc @sahalali 
We may need to update the backend to accomodate required fields for custom taxonomies.Currently we don't have an option to add custom taxonomies as required fields in the formFields.

As per discussion,
Add the list of adminOnlyFields and isRequiredFields as an array inside the forms.

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil 
I am pausing this issue and starting to work in https://github.com/culturecreates/footlight-calendar-api/issues/780 .I need migrate from formschema to formfields in event form first,otherwise I would have to handle both forms and formschema in the mandatory fields section.

**Comment by AbhishekPAnil** [comment]
@troughc Could you add the translations for the mandatory fields section into the CMS copy .Thanks!

**Comment by troughc** [comment]
@AbhishekPAnil please see the spreadsheet. I commented you.

**Comment by AbhishekPAnil** [comment]
Frontend estimate for changing the required fields section only : 21.5H
1.Update the event form - 5
2.Update the person form - 4
3.Update the Organization form - 2.5
4.Update the place form - 5
5.Testing - 5

**Comment by dev-aravind** [comment]
@AbhishekPAnil I updated the calendar apis to consider the mandatoryFields and also updated the template. Please let me know if you need any more changes.

**Comment by troughc** [comment]
NOTE: The Admin only button in taxonomies is not working for standard fields. Please make sure it is working in the back end. so that admins can make standard fields admin only.

**Comment by dev-aravind** [comment]
@AbhishekPAnil This is fixed in the branch 635 right now.

**Comment by troughc** [comment]
@AbhishekPAnil unfortunately I cannot update the figma and I don't love the english text. Please use this text for both the english and french versions of this part of the UI. Thanks!

### English and French Translations for Make calendar read-only section of user settings
Make calendar read-only -> Mettre le calendrier en lecture seule
Turn on to disable editing in this calendar. -> Activez cette option pour désactiver l'édition de ce calendrier.
Read-only -> Lecture seule

**Comment by troughc** [comment]
one more thing @AbhishekPAnil  
Here is exceptional default text for the language selection area. It should work with the bubble approach or the dropdown single select approach):
English-French
Choose languages to display on website
Choisir les langues à afficher sur le site web

**Comment by AbhishekPAnil** [comment]
As per discussion,
Add access in api for admins to edit the calendar metadata
<html>
<body>
<google-sheets-html-origin><style type="text/css"></style>
Update calendar metadata | No | PATCH | calendar-metadata/{id}
-- | -- | -- | --



</body>
</html> 

cc @AbhishekPAnil

**Comment by AbhishekPAnil** [comment]
Test cases for calendar settings

1. Check if all the calendar settings are properly updated upon saving.
2. Check if the change in calendar language is reflected throughout the application.
3. Change the image aspect ratios and max widths and see the changes are reflected across the image uploads in the forms.(Event,Place,Person,Organization).
4. Change the logo and check if the logo is changed in the sidebar and in the widget settings page.
5. Enable/Disable the read only mode and check if the read only mode is correctly maintained in the CMS.
6. Change the event and search template urls and see if its reflected in the widget settings.
7. Try to add taxonomies as filter from the taxonomy forms.

-      See if the add taxonomies are visibile in the filter personalisation dropdowns in the calendar settings.
- Select taxonomies and save the calendar settings and see if the selected taxonomy is visible in the listing pages.Select a concept from the checkbox and check if the filter for the respective listing pages work properly.

**Comment by dev-aravind** [comment]
@AbhishekPAnil I saw 2 bugs in the calendar settings page. 

1. Both forms of Image maximum width disappear when I click backspace once in any one of the fields.
2. Both of those fields (Maxwidth, maxheight) is set as null in payload if you change the values. So if you update it, the values disappear when you open calendar setting next time.

**Comment by AbhishekPAnil** [comment]
@troughc 
Here are the list of date formats currently added in the calendar settings.

1. MM/DD/YYYY
2. DD/MM/YYYY
3. YYYY/MM/DD
4. YYYY-MM-DD
5. DD-MM-YYYY
6. MM-DD-YYYY


Please let me know if you want to add any other formats and on how to proceed with date format display.

**Comment by troughc** [comment]
> [@troughc](https://github.com/troughc) Here are the list of date formats currently added in the calendar settings.
> 
> 1. MM/DD/YYYY
> 2. DD/MM/YYYY
> 3. YYYY/MM/DD
> 4. YYYY-MM-DD
> 5. DD-MM-YYYY
> 6. MM-DD-YYYY
> 
> Please let me know if you want to add any other formats and on how to proceed with date format display. Thanks!

@AbhishekPAnil as per our discussion, for this first release we will not include the date format selector in the Calendar settings. Currently we aren't offering this option in the CMS, the widget, and we are not providing a front end display for the site. Please hide the field/dropdown section - if we need it in the future we will bring it back. Thanks!

**Comment by AbhishekPAnil** [comment]
@troughc Could you update me with the warning message that should be shown while updating a calendar.I might have missed it if you already updated it in the CMS Copy.Thanks!

**Comment by troughc** [comment]
@AbhishekPAnil I am not sure i know what you are looking for, what do you mean 'while updating a calendar'? 



Do you mean making it read only? This was added to the issue https://github.com/culturecreates/footlight-app/issues/798 
Copy for popup message when you first enter a calendar that is in read only mode
English:
Calendar in read only mode
The calendar is currently read only. Editing is unavailable for users. Try again later or contact your admin for details.

French:
Calendrier en mode lecture seule
Le calendrier est actuellement en mode lecture seule. Les utilisateurs ne peuvent pas modifier son contenu. Réessayez plus tard ou contactez votre administrateur pour plus de détails.

**Comment by AbhishekPAnil** [comment]
> @AbhishekPAnil I am not sure i know what you are looking for, what do you mean 'while updating a calendar'? 
> 
> 
> 
> Do you mean making it read only? This was added to the issue https://github.com/culturecreates/footlight-app/issues/798 
> Copy for popup message when you first enter a calendar that is in read only mode
> English:
> Calendar in read only mode
> The calendar is currently read only. Editing is unavailable for users. Try again later or contact your admin for details.
> 
> French:
> Calendrier en mode lecture seule
> Le calendrier est actuellement en mode lecture seule. Les utilisateurs ne peuvent pas modifier son contenu. Réessayez plus tard ou contactez votre administrateur pour plus de détails.

I was referring to the warning message that should be shown when an admin updates the calendar settings(for eg: aspect ratio).

**Comment by troughc** [comment]
@AbhishekPAnil let's talk about it tomorrow - I have to know if it only happens with aspect ratio or if the message needs to be more generic.
Also, can you remove CUSTOM as an option from the aspect ratio dropdown? We won't give them a custom option - they have to use one of hte aspect ratios we are offering. We want to try this approach out.

**Comment by AbhishekPAnil** [comment]
These are the available aspect ratio's :
```
[
    '1:1',
    '2:1',
    '2:3',
    '3:1',
    '3:2',
    '4:1',
    '4:3',
    '5:4',
    '16:9',
    '9:16',
    '21:9',
    '9:21',
   ]
```

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil Notes:
Admin only fields are not shown in the mandatory fields section.
Backend should remove the respective admin only taxonomy field from the mandaotryfield once that taxonomy field is set as an adminonly field.

**Comment by AbhishekPAnil** [comment]
cc @troughc @AbhishekPAnil 
As we discussed ,we will start testing this issue without having the pop warning message while the user updates the calendar.

**Comment by troughc** [comment]
must have image dimensions:
1:1
2:1
4:3
5:4
16:9
The rest might be overkill. If we decide to include the rest, put them after these 5 options (below them) because these are the most commonly used.

**Comment by AbhishekPAnil** [comment]
@troughc @dev-aravind 
Please find the link here to test calendar settings and mandatory fields.
http://calendar-settings-test.s3-website.ca-central-1.amazonaws.com/

**Comment by AbhishekPAnil** [comment]
@troughc 
Please refer the above test cases for calendar settings 
https://github.com/culturecreates/footlight-app/issues/635#issuecomment-2044661615

Test cases for mandatory fields:

1. Check if the mandatory fields section is visible to an admin of the calendar.
2. Check if the admin can add a field as mandatory and save it (Check if the added field is showing in the mandatory fields column and not on the available fields column) .
3. Remove a field from the mandatory field column and save it (Check if the removed field is showing in the available fields column and not on the mandatory fields column).
4. Check if the saved the mandatory are reflected in the respective forms and the validation for each of these fields are showing correctly.
5. The admin should be unable to remove default fields from the mandatory fields column. For example : Event title ,Location ,Date in events.

**Comment by AbhishekPAnil** [comment]
Known issue : 
Calendar images are not visible in this http://calendar-settings-test.s3-website.ca-central-1.amazonaws.com/ link and @dev-aravind is working on fixing it.But everything else ready to test.

**Comment by troughc** [comment]
Excited to start testing!

**Comment by troughc** [comment]
Mandatory settings feedback:

- Visual: The centre line looks to be too thick

When I tested on Wednesday all of the mandatory fields tests mentioned above passed, except 5 as I hadn't tested it.
However today when I tried http://calendar-settings-test.s3-website.ca-central-1.amazonaws.com/ link to test again, saving my changes generated this error which is looping: 


![Image](https://github.com/culturecreates/footlight-app/assets/64489312/d72bdffb-73be-43fc-b408-bf440cf52809)

I used the troubleshooting guide from @SyamBabu-M and it got me out of the loop so I could log back in. :)

However as soon as I clicked something, this error appeared (I redid the method above but it keeps happening)


![Image](https://github.com/culturecreates/footlight-app/assets/64489312/6d989ab0-91e4-4b48-899a-564020b9df7d)

Let me know when I can try testing again.

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil @
Change the code in the frontend to support logo.url.uri and logo.original.url in the payload

**Comment by AbhishekPAnil** [comment]
@troughc You can continue testing .The issue is fixed in both frontend and backend.

**Comment by troughc** [comment]
Tests in Chrome in progress:
### Test cases for calendar settings

- Check if all the calendar settings are properly updated upon saving.

1. The name next to the calendar logo (on the left sidebar) did not update to the french version of the calendar name when I switched from English to French UI:

![Image](https://github.com/culturecreates/footlight-app/assets/64489312/505a4709-969f-4773-bc87-a8c37ac24bf6)
 
It updated only after I refreshed the page.

2. **Check if the change in calendar language is reflected throughout the application.** I changed the Signe Laval calendar language from French (only) to French and English (bilingual). PASSED 
However, I noticed that when I check the events that already exist, the language literal on the english tab was a question mark instead of an fr:


![Image](https://github.com/culturecreates/footlight-app/assets/64489312/57031fc4-b0ca-4962-ba7b-9ee69d6c622b)

In Artsdata the above example event has a name with a french literal: https://kg.artsdata.ca/entity.html?uri=http%3A%2F%2Fkg.artsdata.ca%2Fresource%2FK23-1056



3. Change the image aspect ratios and max widths and see the changes are reflected across the image uploads in the forms.(Event,Place,Person,Organization): PASSED
However, I noticed that when I am using the dropdown to select my aspect ratio, the little down chevron gets replaced with an x when I hover over it. So it clears the field instead of opening the dropdown. I'm not sure what the solution is beause we want to be able to clear the field....The reality is, I have to hover over the field (not the little down chevron) to open the dropdown. Can we make this a little better?

4. Change the logo and check if the logo is changed in the sidebar and in the widget settings page: 
See comment (1) and also the widget image is broken:


![Image](https://github.com/culturecreates/footlight-app/assets/64489312/3e533eb8-ec31-4939-9b19-99a4288d50a3)



5. Enable/Disable the read only mode and check if the read only mode is correctly maintained in the CMS: PASSED
**Question:** I cannot open a Taxonomy, I can only view the list (when in readonly), it that correct? Also, i have full access to ALL settings in read only - is that what we decided?

6. Change the event and search template urls and see if its reflected in the widget settings: PASSED
**Bug discovered:** When I first look show the calendar widget page, the preview is empty and no filters are selected. The preview should show all events. Then i added a filter and events appear in the preview. Then I remove the filter and all events are shown in hte preview

7. Try to add taxonomies as filter from the taxonomy forms: PASSED
8. See if the add taxonomies are visibile in the filter personalisation dropdowns in the calendar settings: PASSED
9. Select taxonomies and save the calendar settings and see if the selected taxonomy is visible in the listing pages: PASSED
10. Select a concept from the checkbox and check if the filter for the respective listing pages work properly. PASSED
11. Refresh the page and check if the filters are persisting: PASSED

### Test cases for mandatory fields:

The centre line looks to be too thick compared to the Figma. Please check and adjust as necessary.

1. Check if the mandatory fields section is visible to an admin of the calendar. 
2. Check if the admin can add a field as mandatory and save it (Check if the added field is showing in the mandatory fields column and not on the available fields column) .
3. Remove a field from the mandatory field column and save it (Check if the removed field is showing in the available fields column and not on the mandatory fields column).
4. Check if the saved the mandatory are reflected in the respective forms and the validation for each of these fields are showing correctly.
5. The admin should be unable to remove default fields from the mandatory fields column. For example : Event title ,Location ,Date in events.

**Comment by troughc** [comment]
@AbhishekPAnil I noticed that some of the French translations aren't there. I am adding them to the copy document and here if possible.  Here it will be <english text> : <french equivalent text>

In Calendar settings french UI:
1. Calendar name : Nom du calendrier

2. The dropdown contents for Fuseau horaire. @saumier can you please help with this?  but my web search only found the following 6:
des fuseaux horaires du Pacifique, des Rocheuses, du Centre, de l'Est, de l'Atlantique et de Terre-Neuve
In the dropdown in the CMS there are these 8 in English:
![Image](https://github.com/culturecreates/footlight-app/assets/64489312/a7e0069e-2760-4ef5-9f6f-204254d90139)

3. Image maximum width : Largeur maximun de l'image
4. Calendar Logo : Logo du calendrier
**Note:** In the logo section, you have an extra sentence in both french and english to remove (the first sentence - Images should be landscape or square and at least 600px wide. )

In Mandatory settings French UI:
Event : Événement
Organization : Organisation
Place : Lieu

**Comment by AbhishekPAnil** [comment]
> 2\. However, I noticed that when I check the events that already exist, the language literal on the english tab was a question mark instead of an fr:

@SyamBabu-M I would need your help on this.



> 3\. However, I noticed that when I am using the dropdown to select my aspect ratio, the little down chevron gets replaced with an x when I hover over it. So it clears the field instead of opening the dropdown. I'm not sure what the solution is beause we want to be able to clear the field....The reality is, I have to hover over the field (not the little down chevron) to open the dropdown. Can we make this a little better?

I have removed the clear icon.I think we don't need this because the user selecting an aspect ratio is mandatory.



> 6\. Change the logo and check if the logo is changed in the sidebar and in the widget settings page:
>     See comment (1) and also the widget image is broken:

This is happening because of mixed content.Our test page is serving in http and the widget is serving in https. But @SyamBabu-M I would like you to test and confirm that when a user changes the calendar logo it is automatically reflected in the widget.

 

> 5\. **Question:** I cannot open a Taxonomy, I can only view the list (when in readonly), it that correct? Also, i have full access to ALL settings in read only - is that what we decided?

@troughc Yes ,in read only mode the admin user can only see the taxonomy listing page.He won't be able to edit any taxonomies and yes you would have full access to the settings pages in read only mode.

> 6\. When I first look show the calendar widget page, the preview is empty and no filters are selected. The preview should show all events. Then i added a filter and events appear in the preview. Then I remove the filter and all events are shown in hte preview

This also happening because of mixed content.

Rest of the fixes are updated in the same testing link.

**Comment by saumier** [comment]
@saumier To pick a standard naming for timezones (in en and fr) and assign back to @AbhishekPAnil

**Comment by saumier** [comment]
### Time Zones in Canada

BTW, the idea of time zones was invented by Scottish Canadian railway engineer Sandford Fleming, who pioneered the use of the 24-hour clock, the world's time zone system, and a standard prime meridian.[2]

This is a very good summary in Wikipedia https://en.wikipedia.org/wiki/Time_in_Canada

There are 6 time zones in Canada, but if you take into account that some regions shift the clock an hour forward for daylight savings in the summer and some don't, then for about half the year there are 10 regions to consider.
https://en.wikipedia.org/wiki/Time_in_Canada#/media/File:Canada_time_zone_map_-_en.svg


For coding, here is the spec from the ECMA standards body:
ECMAScript® 2023 Internationalization API Specification
https://402.ecma-international.org

The ECMAScript 2023 Internationalization API Specification identifies time zones using the Zone and Link names of the IANA Time Zone Database.
https://www.iana.org/time-zones

But rather than download the official Time Zone Database, you can view them in a table in Wikipedia here
[List of tz database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)

I have extracted the related ones in this table.

 TZ Identifier |  Abbreviation Standard Time | Abbreviation Daylight Time
-- | -- | -- |
Canada/Atlantic  |  AST | ADT 
Canada/Central | CST |  CDT
Canada/Eastern |  EST | EDT 
Canada/Mountain |  MST | MDT  
Canada/Newfoundland |  NST |  NDT 
Canada/Pacific |  PST | PDT
Canada/Saskatchewan | CST | -- 
Canada/Yukon |  MST | --

**Comment by saumier** [comment]
Please make any further comments about timezones in #1101

**Comment by saumier** [comment]
@AbhishekPAnil I am assigning back to you. Some of the labels may still change in the future pending further discussion, but that shouldn't hold you up. Please go ahead and use the English and French UI labels in the table above.

**Comment by AbhishekPAnil** [comment]
Please use think new link to test this feature :
https://pr-1103.dla06mbe0wyl7.amplifyapp.com/

**Comment by SyamBabu-M** [comment]
> > 2. However, I noticed that when I check the events that already exist, the language literal on the english tab was a question mark instead of an fr:
> 
> [@SyamBabu-M](https://github.com/SyamBabu-M) I would need your help on this.

@AbhishekPAnil When the calendar is changed from mono-lingual to bilingual, the language fallbacks are not updated. For each language in a calendar, the language fallback must contain an object which acts as the fallback for that language only.
In the situation mentioned by @troughc even though she updated the calendar to bilingual, no fallbacks was added for the newly introduced language. Thus causing the literal to show "?" symbol.

**Comment by troughc** [comment]
French UI fixes: PASSED
- Calendar name : Nom du calendrier: PASSED
- The dropdown contents for Fuseau horaire: PASSED 
- Image maximum width : Largeur maximale: PASSED
- Calendar Logo : Logo du calendrier: PASSED
- Mandatory fields french language fixes: PASSED

Filter personalization test: PASSED


**To do:**
1. Calendar Logo : you have an extra sentence in both french and english to remove 
- ENGLISH: REMOVE the first sentence - Images should be landscape or square and at least 600px wide. )
- FRENCH: Remove "Les images doivent être au format paysage ou carré et mesurer au moins 600 px de large."

2. Image update issue: the display in the modal is not updated the first time you click to edit an image (large image isn't correct aspect ratio). However after you do it once, it is fine, even is you cancel (display issue).

3. Logo image (small thumbnail below image in calendar settings) is stretched - please correct

4. Mandatory fields not staying updated: Please recheck the mandatory fields as discussed

5. There is an issue with setting the Admin only field in Taxonomies. As discussed, the fix needs to be implemented.

**Comment by troughc** [comment]
@AbhishekPAnil FYI

![Image](https://github.com/culturecreates/footlight-app/assets/64489312/7cc63471-b965-418e-81c8-b960a77ca61d)

 i just tried to remove mandatory fields from Signe Laval, and I got kicked out and can't get out of the loop. (would have to use Syam's fix)

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil 
From the above comment .
(5) is happening only in French calendar.

Fix the edit icon being outside the box.Fix the stretching of the smaller image.


![Image](https://github.com/culturecreates/footlight-app/assets/117100211/3fa9982f-7314-49da-b920-a7b0e5dcec33)

**Comment by troughc** [comment]
also, (2) is working correctly now. 
@AbhishekPAnil I rechecked the aspect ratio of the image on the listing page (I tested a few differen sizes) and it looks like it is correct. So nothing to do there.

**Comment by AbhishekPAnil** [comment]
@troughc I have fixed the issue where a user with interface language french was unable to save the mandatory fields.The fix is updated in the same link.

**Comment by troughc** [comment]
@AbhishekPAnil there seems to be a word count requirement when I add description for events, and the word count is unknown (and high):
<img width="533" alt="Screenshot 2024-05-14 at 1 32 11 PM" src="https://github.com/culturecreates/footlight-app/assets/64489312/18ed68a4-7f40-4d38-9510-7b7d16c35a63">

I tested this on the DICI calendar - I also tested on Tout Culture and Signe Laval
Steps to reproduce:
Signe Laval had description required so it worked with two words. 
I then removed event description as a requirement. 
The first time went into an event WITH a description after removing the requirement, I deleted the description. Red text 
was displayed saying I need a description and it was too short. I saved as draft and then reentered the event. The red text was gone and I could publish without a description (second visit to event).
I then added description as a mandatory field. When I try to add a description it seems to require a very long description,

Hope this is clear enough.

**Comment by AbhishekPAnil** [comment]
> @AbhishekPAnil there seems to be a word count requirement when I add description for events, and the word count is unknown (and high):
> <img width="533" alt="Screenshot 2024-05-14 at 1 32 11 PM" src="https://github.com/culturecreates/footlight-app/assets/64489312/18ed68a4-7f40-4d38-9510-7b7d16c35a63">
> 
> I tested this on the DICI calendar - I also tested on Tout Culture and Signe Laval
> Steps to reproduce:
> Signe Laval had description required so it worked with two words. 
> I then removed event description as a requirement. 
> The first time went into an event WITH a description after removing the requirement, I deleted the description. Red text 
> was displayed saying I need a description and it was too short. I saved as draft and then reentered the event. The red text was gone and I could publish without a description (second visit to event).
> I then added description as a mandatory field. When I try to add a description it seems to require a very long description,
> 
> Hope this is clear enough.
> 
> 
> 
> 
> 
@troughc 
I have a doubt,when we add a description as a mandatory field ,how will the admin add a word count? Initially we can set it in the backend,but when the admin comes in and removes the description field from mandatory fields the data of minimum word count is kind of lost.So when he comes back in and  add the description as a mandatory field,we dont know what must be set as the minimum word count!

**Comment by troughc** [comment]
> @troughc I have fixed the issue where a user with interface language french was unable to save the mandatory fields.The fix is updated in the same link PASSED

**Comment by troughc** [comment]
> @troughc
> I have a doubt,when we add a description as a mandatory field ,how will the admin add a word count? Initially we can set it in the backend,but when the admin comes in and removes the description field from mandatory fields the data of minimum word count is kind of lost.So when he comes back in and add the description as a mandatory field,we dont know what must be set as the minimum word count!

I think the word count must be set to 1 word for the moment (like it is for Signe Laval).

**Comment by troughc** [comment]
Consolidating test results here @AbhishekPAnil 
Calendar Logo : you have an extra sentence in both french and english to remove: PASSED
Image update issue: the display in the modal is not updated the first time you click to edit an image (large image isn't correct aspect ratio). However after you do it once, it is fine, even is you cancel (display issue). PASSED
Mandatory fields not staying updated: Please recheck the mandatory fields as discussed (was in French only): PASSED
taxonomy admin only field display problem is now covered in another issue: https://github.com/culturecreates/footlight-app/issues/1108


Left to do:

- Fix the edit icon being outside the box.Fix the stretching of the smaller image.
- Fix the word count issue (and check for all entities)
- Anything else?

Thanks!

**Comment by AbhishekPAnil** [comment]
@troughc I think we can track the image stretching and edit being outside as a separate issue.This happens for the other workspaces only.

**Comment by AbhishekPAnil** [comment]
cc @AbhishekPAnil 
The minimum word count is set as "1" when an admin revisits and sets the description as mandatory from the mandatory fields section.

**Comment by troughc** [comment]
@AbhishekPAnil Tammy found a couple new issues:
1. she could add Event accessibility as a Mandatory field even though she had NOT created the taxonomy for it. Therefore Event Accessiblity is NOT in the form (it doesn't exist as a property in Events). When she then clicks Save (as published) this error shows up.

![Image](https://github.com/culturecreates/footlight-app/assets/64489312/8ce828d6-6489-401c-af3e-216dc3533ae1)


**Expected behavior:** no field should be available as a Mandatory field if it does NOT appear in the form. In this example, Event Accessiblity should NOT be available as a Mandatory field, until it actually exists (until a taxonomy has been created and it is NOT an Admin only field)

2. Text that is blue on the side (fields like Facebook event, keywords, etc) can be set as Mandatory fields. When this happens they should be moved to the main form. This is not happening. As a result, when Tammy didn't fill the facebook event field, she got a message that information was missing, but couldn't see where. 
**Expected behavior:** All Mandatory fields should be displayed in the main form.

**Comment by AbhishekPAnil** [comment]
> 2\. Text that is blue on the side (fields like Facebook event, keywords, etc) can be set as Mandatory fields. When this happens they should be moved to the main form. This is not happening. As a result, when Tammy didn't fill the facebook event field, she got a message that information was missing, but couldn't see where.

 I am unable to reproduce this issue. Do you remember which calendar/event you were testing on or the steps to recreate this ?Thanks!

**Comment by AbhishekPAnil** [comment]
> 2\. Text that is blue on the side (fields like Facebook event, keywords, etc) can be set as Mandatory fields. When this happens they should be moved to the main form. This is not happening. As a result, when Tammy didn't fill the facebook event field, she got a message that information was missing, but couldn't see where.
>     **Expected behavior:** All Mandatory fields should be displayed in the main form.

The task was already implement.This was a bug which was happening only for existing events.This wasn't happening when a user tries to create a new event.I have fixed the bug and updated in the same link.

**Comment by dev-aravind** [comment]
@troughc Both if these bugs are now fixed. You can test it [here](https://pr-1103.dla06mbe0wyl7.amplifyapp.com/dashboard/6308ef4a7f771f00431d939a/events).

**Comment by troughc** [comment]
@dev-aravind @AbhishekPAnil  (2) the mandatory fields; 
This is not fixed for all fields in existing events. the one i noticed is in Places:
- Opening hours link doesn't open when made mandatory for existing events.

Not all fields that are made mandatory get the star. Examples: 
- Description field in Place form
- Organizations form looks okay
- People form looks okay
- Events form: performer field, supporter field
To reproduce, make all fields mandatory then look at an event (existing event, then new event)
Finally , when testing Opening hours link mandatory for Places, it looked like this (the title is wrong). Maybe it's a separate bug...:

![Image](https://github.com/culturecreates/footlight-app/assets/64489312/b431aa65-caec-4b81-b6bc-3eee420260a0)

**Comment by AbhishekPAnil** [comment]
> * Opening hours link doesn't open when made mandatory for existing events.

> * Description field in Place form

> * Events form: performer field, supporter field

Sorry for this,I forgot to add the condition for checking if its mandatory in the code.This is fixed and updated in the link.

> *when testing Opening hours link mandatory for Places, it looked like this (the title is wrong). Maybe it's a separate bug...:

This happened while resolving a merge conflict.This is also fixed and updated in the link.

**Comment by troughc** [comment]
@SyamBabu-M I tested the checkbox issue with the Admin only field. It is working now.

**Comment by troughc** [comment]
Still pending a fix is the removing admin only fields from the mandatory fields list
Here is what is happening today (we are getting duplicates in the list, I think because I am turning Admin only fields on and off. (I tried it several times).


![Image](https://github.com/culturecreates/footlight-app/assets/64489312/046529e3-b8cb-457e-92f2-041919a9f428)



![Image](https://github.com/culturecreates/footlight-app/assets/64489312/248652fb-d90e-49dc-a144-86ec8321e65a)

**Comment by dev-aravind** [comment]
@troughc The bugs that you mentioned are fixed in the test server. Please review it and let me know if you see any other issues.

**Comment by troughc** [comment]
Tested the bugs mentioned (admin only field and mandatory fields): PASSED. I think this is ready to go. @saumier can you please do a light pass? If not we can push this to deploy.

**Comment by troughc** [comment]
one caveat @AbhishekPAnil and @dev-aravind, it seems if admin only is already set in the taxonomy (before I modify it), that taxonomly is appearing in the Mandatory field list. However, after I uncheck mandatory field and recheck it, everything is good. I left a couple examples in Signe Laval: See Type de lieux taxonomy (admin only) in Places and then look at the mandatory fields.
If you uncheck admin only and save, then go back in and check Admin only and save, the mandatory fields will no longer have type de lieux.

cc @saumier

**Comment by saumier** [comment]
@troughc  I did a quick run through and it looks good.

**Comment by troughc** [comment]
@AbhishekPAnil I just did a last run of tests and I didn't find any issues. yay!

**Comment by dev-aravind** [comment]
@troughc There was a small fix related to the getCalendarBySlug API in the Open-API. It's fixed in the staging and production environment as of now,
